aws/internal/service/efs/waiter/status.go

```diff
@@ -0,0 +1,30 @@
+package waiter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/efs"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// AccessPointStatus fetches the Access Point and its Status
+func AccessPointStatus(conn *efs.EFS, accessPointId string) resource.StateRefreshFunc {
```

Comment:
Attribute is named `LifeCycleState`

```suggestion
// AccessPointLifeCycleState fetches the Access Point and its LifecycleState
func AccessPointLifeCycleState(conn *efs.EFS, accessPointId string) resource.StateRefreshFunc {
```